src/app/cli/src/commands/system_info_command.rs

```diff
@@ -164,3 +184,58 @@ impl BuildInfo {
         }
     }
 }
+
+#[derive(Debug, serde::Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct WorkspaceInfo {
+    pub version: u32,
+    pub root_dir: String,
+}
+
+impl WorkspaceInfo {
+    pub fn collect(workspace_svc: Arc<WorkspaceService>) -> Self {
+        if !workspace_svc.is_in_workspace() {
+            return Self {
+                version: 0,
+                root_dir: "".to_string(),
+            };
+        }
+        let root_dir = match workspace_svc.layout() {
+            Some(wl) => wl.root_dir.to_str().map(String::from).unwrap(),
+            None => "".to_string(),
+        };
+        Self {
+            version: workspace_svc.workspace_version().unwrap().unwrap().into(),
```

Comment:
You should consider that `system info` is a command that users will run when something is going seriously wrong and their workspace might be in a bad state.

The last thing we want is for this command to crash and provide us zero information.

So I would be extremely careful with `unwrap()`s here and provide fallbacks everywhere. One missing value is OK, but crashing would be unacceptable.